### DISCUSSION TITLE
Print out the history of iteration number of nonlinear solver

### DISCUSSION THREAD

**Discussion by tqcthai** [idaholab/moose#25505] [Q&A General] [answered]
Hello Moose community,

Do you know how to print out the number of Newton iteration in nonlinear solver in a final report?
For instance, 

at time step 1, the solver is converged after m1 steps,
at time step 2, the solver is converged after m2 steps,
... 
at time step n, the solver is converged after mn steps.

And I need to have the information of m1, m2,... and mn in an output report. I wonder how to do it?

**Comment by GiudGiud** [comment, marked as answer]
use this https://mooseframework.inl.gov/source/postprocessors/NumNonlinearIterations.html
it will be available in a CSV file after

**Reply by tqcthai** [reply]
It works well. Thank you so much.
